urld0f681c558_farom57_AstroidDriver_tgz-pJ8-laazotea_indi_server_examples_MinimumINDIServerJ8/src/laazotea/indi/INDISexagesimalFormatter.java

```diff
@@ -0,0 +1,355 @@
+ * 
+ *  INDI for Java is free software: you can redistribute it
+ *  and/or modify it under the terms of the GNU General Public License 
+ *  as published by the Free Software Foundation, either version 3 of 
+ *  the License, or (at your option) any later version.
+ * 
+ *  INDI for Java is distributed in the hope that it will be
+ *  useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+ *  of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ * 
+ *  You should have received a copy of the GNU General Public License
+ *  along with INDI for Java.  If not, see 
+ *  <http://www.gnu.org/licenses/>.
+ */
+package laazotea.indi;
+
+import java.util.Formatter;
+import java.util.Locale;
+import java.util.NoSuchElementException;
+import java.util.StringTokenizer;
+
+/**
+ * A class to format and parse numbers in sexagesimal format.
+ *
+ * @author S. Alonso (Zerjillo) [zerjioi at ugr.es]
+ * @version 1.10, March 19, 2012
+ */
+public class INDISexagesimalFormatter {
+
+  private String format;
+  private int length;
+  private int fractionLength;
+
+  /**
+   * Constructs an instance of
+   * <code>INDISexagesimalFormatter</code> with a particular format.
+   *
+   * @param format The desired format
+   * @throws IllegalArgumentException if the format is not correct: begins with
+   * %, ends with m and specifies a length and fractionLength in the form
+   * length.fractionLength. Valid fractionLengths are 3, 5, 6, 8 and 9. For
+   * example %5.3m.
+   */
+  public INDISexagesimalFormatter(String format) throws IllegalArgumentException {
+    this.format = format;
+
+    checkFormat();
+  }
+
+  /**
+   * Gets the format of this formatter.
+   *
+   * @return the format of this formatter.
+   */
+  public String getFormat() {
+    return format;
+  }
+
+  /**
+   * Checks the specified format string.
+   *
+   * @throws IllegalArgumentException if the format string is not valid: begins
+   * with %, ends with m and specifies a length and fractionLength in the form
+   * length.fractionLength. Valid fractionLengths are 3, 5, 6, 8 and 9. For
+   * example %5.3m.
+   */
+  private void checkFormat() throws IllegalArgumentException {
+    if (!format.startsWith("%")) {
+      throw new IllegalArgumentException("Number format not starting with %");
+    }
+
+    if (!format.endsWith("m")) {
+      throw new IllegalArgumentException("Sexagesimal format not recognized (not ending m)");
+    }
+
+    String remaining = format.substring(1, format.length() - 1);
+
+    int dotPos = remaining.indexOf(".");
+
+    if (dotPos == -1) {
+      throw new IllegalArgumentException("Sexagesimal format not correct (no dot)");
+    }
+
+    String l = remaining.substring(0, dotPos);
+    String frLength = remaining.substring(dotPos + 1);
+
+    try {
+      length = Integer.parseInt(l);
+      fractionLength = Integer.parseInt(frLength);
+    } catch (NumberFormatException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)